addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,327 @@
+                } else if (command.equals("power_off")) {
+                    power = false;
+                    updateState(getThing().getChannel("mute").getUID(), OnOffType.OFF);
+                    updateState(getThing().getChannel("power").getUID(), OnOffType.OFF);
+                    updateState(getThing().getChannel("volume").getUID(), UnDefType.NULL);
+                    updateState(getThing().getChannel("source").getUID(), UnDefType.NULL);
+                } else if (command.equals("power_on")) {
+                    power = true;
+                    updateState(getThing().getChannel("power").getUID(), OnOffType.ON);
+                    powerOnRefresh();
+                } else if (command.equals("power")) {
+                    String state = readUntil('!');
+                    if (state.equals("on")) {
+                        power = true;
+                        updateState(getThing().getChannel("power").getUID(), OnOffType.ON);
+                        powerOnRefresh();
+                    } else if (state.equals("standby")) {
+                        updateState(getThing().getChannel("mute").getUID(), OnOffType.OFF);
+                        updateState(getThing().getChannel("power").getUID(), OnOffType.OFF);
+                        power = false;
+                    }
+                } else if (command.equals("dimmer")) {
+                    updateState(getThing().getChannel("dimmer").getUID(), readDimmer());
+                } else if (command.equals("freq")) {
+                    updateState(getThing().getChannel("frequency").getUID(), readFrequency());
+                } else if (command.equals("source")) {
+                    updateState(getThing().getChannel("source").getUID(), new StringType(readUntil('!')));
+                } else if (command.equals("display")) {
+                    String str_length = readUntil(',');
+                    int length = Integer.parseInt(str_length);
+                    byte[] data = new byte[length];
+                    int off = 0;
+                    while (off != length) {
+                        int r = serialPort.getInputStream().read(data, off, length - off);
+                        if (r == -1) {
+                            throw new IOException("Connection closed while reading display content");
+                        } else {
+                            off += r;
+                        }
+                    }
+                    // We don't do anything with display content, could add a channel if this is useful
+                } else {
+                    readUntil('!'); // discard
+                }
+
+            } catch (IOException e) {
+                logger.error("Input error while receiving data from amplifier", e);
+                disconnect();
+            }
+        }
+    }
+
+    void send(String text) throws IOException {
+        if (power) {
+            connect();
+            serialPort.getOutputStream().write(text.getBytes());
```

Comment:
A lot lines up you called getBytes with a charset, please do that everywhere, its best-practice to do so. 

As mentioned above please extract `serialPort.getOutputStream().write(text.getBytes("ascii"));` to a separate method.